File: src/main/java/wiles/parser/statements/expressions/InnerExpression.java
package wiles.parser.statements.expressions;

import wiles.parser.builders.Context;
import wiles.shared.Token;
import org.jetbrains.annotations.NotNull;
import wiles.parser.exceptions.TokenExpectedException;
import wiles.parser.exceptions.UnexpectedEndException;

import static wiles.parser.builders.ExpectParamsBuilder.tokenOf;
import static wiles.shared.constants.ErrorMessages.INTERNAL_ERROR;
import static wiles.shared.constants.Tokens.PAREN_END_ID;

public class InnerExpression extends AbstractExpression {
    public InnerExpression(@NotNull Context oldContext) {
        super(oldContext.setWithinInnerExpression(true));
    }

    {
        isInner = true;
    }

    @Override
    protected boolean handleToken(@NotNull Token token) throws TokenExpectedException, UnexpectedEndException {
        if(token.getContent().equals(PAREN_END_ID)) {
            transmitter.expect(tokenOf(PAREN_END_ID).withErrorMessage(INTERNAL_ERROR));
            return true;
        }
        return false;
    }

}

File: wiles-base/src/main/java/wiles/parser/statements/expressions/InnerExpression.kt
package wiles.parser.statements.expressions

import wiles.parser.builders.ExpectParamsBuilder.Companion.tokenOf
import wiles.parser.builders.ParserContext
import wiles.parser.exceptions.TokenExpectedException
import wiles.parser.exceptions.UnexpectedEndException
import wiles.shared.Token
import wiles.shared.constants.ErrorMessages.INTERNAL_ERROR
import wiles.shared.constants.Tokens.PAREN_END_ID

class InnerExpression(oldContext: ParserContext) : AbstractExpression(oldContext) {
    @Throws(TokenExpectedException::class, UnexpectedEndException::class)
    override fun handleToken(token: Token): Boolean {
        if (token.content == PAREN_END_ID) {
            transmitter.expect(tokenOf(PAREN_END_ID).withErrorMessage(INTERNAL_ERROR))
            return true
        }
        return false
    }
}
